src/control/server/telemetry.go

```diff
@@ -76,9 +92,14 @@ func startPrometheusExporter(ctx context.Context, log logging.Logger, port int,
 	listenAddress := fmt.Sprintf("0.0.0.0:%d", port)
 	log.Infof("Listening on %s", listenAddress)
 
-	if err := http.ListenAndServe(listenAddress, nil); err != nil {
-		return err
-	}
+	// http listener is a blocking call
+	go func() {
+		err := http.ListenAndServe(listenAddress, nil)
```

Comment:
Hmm, not sure how important graceful shutdown is for this case... That's more important for a service where you want to avoid terminating in the middle of a request. In our case I don't think it's a big deal if we bomb out in the middle of a telemetry scrape, particularly because we're in kind of a weird state by that point with lots of stuff already shut down.

Reply:
Yeah, I wasn't sure what to do in this case. I don't know that there's much to clean up, but I could do a minor refactor to let us shutdown gracefully in the cleanup function.